src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -70,14 +68,14 @@ int D4NFilterDriver::initialize(CephContext *cct, const DoutPrefixProvider *dpp)
 
   cacheDriver->initialize(cct, dpp);
 
-  objDir->init(cct);
-  blockDir->init(cct);
+  objDir->init(cct, dpp);
+  blockDir->init(cct, dpp);
 
-  policyDriver->init();
-  policyDriver->cachePolicy->init(cct);
+  policyDriver->init(); 
+  policyDriver->get_cache_policy()->init(cct, dpp);
 
   lruPolicyDriver->init();
-  lruPolicyDriver->cachePolicy->init(cct);
+  lruPolicyDriver->get_cache_policy()->init(cct, dpp);
```

Comment:
You can remove lru policy related code from d4n filter driver, it was just a prototype. It can be plugged back in when work on write-back cache starts.